backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -167,15 +168,20 @@ private void createRoleBindingsForDataset(Dataset dataset, UserInfo userInfo) {
         new CollaboratorUser(authService, userInfo),
         dataset.getId(),
         ModelDBServiceResourceTypes.DATASET);
-    if (dataset.getDatasetVisibility().equals(DatasetVisibility.PUBLIC)) {
+
+    DatasetVisibility datasetVisibility;
+    if (App.getInstance().getPublicSharingEnabled()) {
+      datasetVisibility = DatasetVisibility.PUBLIC;
+    } else {
+      datasetVisibility = dataset.getDatasetVisibility();
+    }
+
```

Comment:
Can we move this logic in `createDataset` and this function will work independently and only take responsibility to create  RoleBinding based on the dataset?

Reply:
done